homeassistant/helpers/entity_platform.py

```diff
@@ -792,6 +757,61 @@ def remove_entity_cb() -> None:
 
         await entity.add_to_platform_finish()
 
+    @callback
+    def _async_process_device_info(
+        self, device_info: DeviceInfo
+    ) -> dev_reg.DeviceEntry | None:
+        """Process a device info."""
+        keys = set(device_info)
+
+        # If no keys or not enough info to match up, abort
+        if len(keys & {"connections", "identifiers"}) == 0:
+            self.logger.error(
+                "Ignoring device info without identifiers or connections: %s",
+                device_info,
+            )
+            return None
```

Comment:
It didn't raise in the old code either, but is there a good reason to not raise here? Why do we go ahead and add an entity which has invalid device info?

Reply:
I chose not to change that behavior as it would be breaking. It would probably be a small breaking change as I was pleasantly surprised the low amount of bad device info this PR found. If we decide to do this, we should do it in a follow-up PR.